state/memory_test.go

```diff
@@ -0,0 +1,40 @@
+package state
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+func TestInMemoryDb_SelfDestruct6780OnlyDeletesContractsCreatedInSameTransaction(t *testing.T) {
+	a := common.Address{1}
+	b := common.Address{2}
+
+	db := MakeInMemoryStateDB(nil, 12)
+	db.CreateContract(a)
```

Comment:
I think you should put `b` into the db. You can create a `WorldState` and pass it as first argument of the factory.

Reply:
It would add these lines to the code, without affecting anything in the test:
```
ws := txcontext.NewWorldState(map[common.Address]txcontext.Account{
      b: txcontext.NewAccount(nil, nil, nil, 1),
})
db := MakeInMemoryStateDB(ws, 12)
```
@petr-hanzl To you prefer this to be added?

Reply:
I felt like we should add the account to the database so that we see that `db.Selfdestruct6780(b)` does not destroy it. But now that I dig into the code of `inMemoryDB`, it does not make sense.